adapters_test.go

```diff
@@ -0,0 +1,355 @@
+package monomer_test
+
+import (
+	"testing"
+
+	bfttypes "github.com/cometbft/cometbft/types"
+	codectypes "github.com/cosmos/cosmos-sdk/codec/types"
+	sdktx "github.com/cosmos/cosmos-sdk/types/tx"
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	ethtypes "github.com/ethereum/go-ethereum/core/types"
+	"github.com/kataras/iris/v12/x/errors"
+	"github.com/polymerdao/monomer"
+	"github.com/polymerdao/monomer/testutils"
+	rolluptypes "github.com/polymerdao/monomer/x/rollup/types"
+	"github.com/stretchr/testify/require"
+)
+
+func TestAdaptPayloadTxsToCosmosTxs(t *testing.T) {
+	t.Run("nil tx", func(t *testing.T) {
+		txs, err := monomer.AdaptPayloadTxsToCosmosTxs(nil, nil, "")
+		require.NoError(t, err)
+		require.Empty(t, txs)
+	})
+
+	t.Run("0 txd", func(t *testing.T) {
+		txs, err := monomer.AdaptPayloadTxsToCosmosTxs([]hexutil.Bytes{}, nil, "")
+		require.NoError(t, err)
+		require.Empty(t, txs)
+	})
+
+	simpleSigner := func(_ *sdktx.Tx) error {
+		return nil
+	}
+
+	tests := []struct {
+		name, from            string
+		depTxNum, nonDepTxNum int
+		signTx                monomer.TxSigner
+	}{
+		{
+			name:     "1 dep tx",
+			depTxNum: 1,
+		},
+		{
+			name:        "1 + 1 txs",
+			depTxNum:    1,
+			nonDepTxNum: 1,
+		},
+		{
+			name:        "10 + 10 txs",
+			depTxNum:    10,
+			nonDepTxNum: 10,
+		},
+		{
+			name:        "3 + 3 txs + from",
+			depTxNum:    3,
+			nonDepTxNum: 3,
+			from:        "from",
+		},
+		{
+			name:        "3 + 3 txs + signTx",
+			depTxNum:    3,
+			nonDepTxNum: 3,
+			signTx:      simpleSigner,
+		},
+		{
+			name:        "3 + 3 txs + from + signTx",
+			depTxNum:    3,
+			nonDepTxNum: 3,
+			from:        "from",
+			signTx:      simpleSigner,
+		},
+	}
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			inclusionListTxs, _, txBytes := generateTxsAndTxsBytes(t, test.depTxNum, test.nonDepTxNum)
+			txs, err := monomer.AdaptPayloadTxsToCosmosTxs(txBytes, test.signTx, test.from)
+			require.NoError(t, err)
+
+			msgAny, err := codectypes.NewAnyWithValue(&rolluptypes.MsgApplyL1Txs{
+				TxBytes:     inclusionListTxs[:test.depTxNum],
+				FromAddress: test.from,
+			})
+			require.NoError(t, err)
+
+			sdkTx := sdktx.Tx{
+				Body: &sdktx.TxBody{
+					Messages: []*codectypes.Any{msgAny},
+				},
+			}
+
+			if test.signTx != nil {
+				err := test.signTx(&sdkTx)
+				require.NoError(t, err)
+			}
+
+			depositSDKMsgBytes, err := sdkTx.Marshal()
+			require.NoError(t, err)
+
+			cosmosTxs := make(bfttypes.Txs, 0, 1+test.nonDepTxNum)
+			cosmosTxs = append(cosmosTxs, depositSDKMsgBytes)
+
+			for _, cosmosTx := range inclusionListTxs[test.depTxNum:] {
+				var tx ethtypes.Transaction
+				err := tx.UnmarshalBinary(cosmosTx)
+				require.NoError(t, err)
+				cosmosTxs = append(cosmosTxs, tx.Data())
+			}
+
+			require.Equal(t, cosmosTxs, txs)
+		})
+	}
+
+	t.Run("unmarshal binary error", func(t *testing.T) {
```

Comment:
The test names should describe the test case, not the error string. "invalid attributes transaction" could be a good name here

Same for the other places.

Reply:
[Rename testcases](https://github.com/polymerdao/monomer/pull/157/commits/274e965c84a6d0e8b03bbff3913111991b632e15)